### PR TITLE
sui::vec_map::keys returns a list of keys in VecMap 

### PR DESCRIPTION
**Pull request by bausano**
Adding API to vec_map which returns a list of keys in it. This allows us to iterate over a vec_map without having to own it.

### CODE REVIEW

File: crates/sui-framework/tests/vec_map_tests.move [hunk 1]
```diff
@@ -121,4 +121,17 @@ module sui::vec_map_tests {
...
+        let m = vec_map::empty();
+
+        assert!(vec_map::keys(&m) == vector::empty(), 0);
+
+        vec_map::insert(&mut m, 1, true);
+        vec_map::insert(&mut m, 5, false);
+
+        let expected_keys = vector::singleton(5);
+        vector::push_back(&mut expected_keys, 1);
+        assert!(vec_map::keys(&m) == expected_keys, 1);
```
Comment (sblackshear) [inline crates/sui-framework/tests/vec_map_tests.move:135]:
```suggestion
        assert!(vec_map::keys(&m) == vector[1, 5], 1);
```

File: crates/sui-framework/tests/vec_map_tests.move [hunk 2]
```diff
@@ -121,4 +121,17 @@ module sui::vec_map_tests {
         }
     }
 
+    #[test]
+    fun return_list_of_keys() {
+        let m = vec_map::empty();
+
+        assert!(vec_map::keys(&m) == vector::empty(), 0);
```
Comment (sblackshear) [inline crates/sui-framework/tests/vec_map_tests.move:128]:
```suggestion
        assert!(vec_map::keys(&m) == vector[], 0);
```

File: crates/sui-framework/tests/vec_map_tests.move (hunk 1 above)
Comment (sblackshear) [reply crates/sui-framework/tests/vec_map_tests.move:135]: This will let you kill the `insert`'s above

**Review by sblackshear — APPROVED** [review]

File: crates/sui-framework/tests/vec_map_tests.move (hunk 1 above)
Comment (bausano) [reply crates/sui-framework/tests/vec_map_tests.move]: thanks!

File: crates/sui-framework/tests/vec_map_tests.move [hunk 3]
```diff
@@ -121,4 +121,17 @@ module sui::vec_map_tests {
...
+    #[test]
+    fun return_list_of_keys() {
+        let m = vec_map::empty();
+
+        assert!(vec_map::keys(&m) == vector[], 0);
+
+        vec_map::insert(&mut m, 1, true);
+        vec_map::insert(&mut m, 5, false);
+
+        let expected_keys = vector::singleton(5);
```
Comment (sblackshear) [inline crates/sui-framework/tests/vec_map_tests.move:133]: `expected_keys` is now unused and can be deleted.

**Comment by sblackshear** [comment]
Note: any changes to the Sui Framework affect genesis, and there are a few expected value tests that need to be updated: `cargo test; cargo insta review` in both `sui-config` and `sui-cost` should do the trick.

**Comment by Ahmedpooh** [comment]
Okay

File: crates/sui-framework/tests/vec_map_tests.move (hunk 3 above)
Comment (bausano) [reply crates/sui-framework/tests/vec_map_tests.move]: ah right, done

**Comment by bausano** [comment]
> Note: any changes to the Sui Framework affect genesis, and there are a few expected value tests that need to be updated: `cargo test; cargo insta review` in both `sui-config` and `sui-cost` should do the trick.

I generated new snapshots and replaced the old ones with them.

**Pull request merged**